crates/storage/src/schema/revision_0051.rs

```diff
@@ -1,114 +1,1746 @@
 
-#[derive(Clone, Default, Debug, Deserialize, Serialize, PartialEq, Eq)]
-#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
-pub enum ExecutionStatus {
-    // This must be the default as pre v0.12.1 receipts did not contain this value and
-    // were always success as reverted did not exist.
-    #[default]
-    Succeeded,
-    Reverted,
-}
+    let mut progress_logged = Instant::now();
+    const LOG_RATE: Duration = Duration::from_secs(10);
 
-/// NOTE: DO NOT RUN THIS IN PROD.
-/// TODO: Update this to use bincode serialization instead of serde_json. (Follow-up PR)
-pub(crate) fn migrate(tx: &rusqlite::Transaction<'_>) -> anyhow::Result<()> {
+    let count = tx.query_row("SELECT COUNT(*) FROM starknet_transactions", [], |row| {
+        row.get::<_, i64>(0)
+    })?;
     tx.execute(
         r"
-ALTER TABLE starknet_transactions ADD COLUMN events BLOB DEFAULT NULL;
-",
+        CREATE TABLE starknet_transactions_new (
+            hash        BLOB PRIMARY KEY,
+            idx         INTEGER NOT NULL,
+            block_hash  BLOB NOT NULL,
+            tx          BLOB,
+            receipt     BLOB,
+            events      BLOB
+        )",
         [],
     )
-    .context("Altering starknet_transactions table: adding new column")?;
-    let mut stmt = tx.prepare("SELECT hash, receipt FROM starknet_transactions")?;
-    let mut rows = stmt.query([])?;
-    let mut compressor = zstd::bulk::Compressor::new(10).context("Create zstd compressor")?;
-    while let Some(row) = rows.next()? {
-        let hash = row.get_ref_unwrap(0).as_blob()?;
-        let receipt = row.get_ref_unwrap(1).as_blob()?;
-        let receipt = zstd::decode_all(receipt).context("Decompressing receipt")?;
-        let mut receipt: ReceiptDto =
-            serde_json::from_slice(&receipt).context("Deserializing receipt")?;
-        let events =
-            serde_json::to_vec(&receipt.events.as_ref().unwrap()).context("Serializing events")?;
-        let events = compressor.compress(&events).context("Compressing events")?;
-        receipt.events = None;
-        let receipt = serde_json::to_vec(&receipt).context("Serializing receipt")?;
-        let receipt = compressor
-            .compress(&receipt)
-            .context("Compressing receipt")?;
-        tx.execute(
-            "UPDATE starknet_transactions SET receipt = ?, events = ? WHERE hash = ?",
-            params![receipt, events, hash],
-        )?;
+    .context("Creating starknet_transactions_new table")?;
+    let mut query_stmt =
+        tx.prepare("SELECT hash, idx, block_hash, tx, receipt FROM starknet_transactions")?;
+    let mut insert_stmt = tx.prepare(
+        r"INSERT INTO starknet_transactions_new (hash, idx, block_hash, tx, receipt, events)
+                                         VALUES (?, ?, ?, ?, ?, ?)",
+    )?;
+    const BATCH_SIZE: usize = 10_000;
+    let mut rows = query_stmt.query([])?;
+    let mut progress = 0;
+    loop {
+        if progress_logged.elapsed() > LOG_RATE {
```

Comment:
I suggest adding in `count % 1000 == 0` otherwise you poll `.elapsed()` on every iteration which is quite expensive (its a syscall).

```suggestion
        if count % 1000 == 0 && progress_logged.elapsed() > LOG_RATE {
```

Reply:
I don't think we need that here. This check runs once for every batch, which is 10k rows.